File: 118-pascals-triangle/pascals-triangle.cpp
class Solution {
public:
    vector<int> pattern(int ind){
        vector<int> row;
        int sum = 1;
        row.push_back(1);
        for(int i=1;i<ind;i++){
            sum *= (ind - i);
            sum /= i;

            row.push_back(sum);
        }
        return row;
    }
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> ans;
        for(int i=1;i<=numRows;i++){
            ans.push_back(pattern(i));
        }
        return ans;
    }
};

File: 118-pascals-triangle/pascals-triangle.java
class Solution {
    public static List<Integer> rows(int n){
        long ans=1;
        List<Integer> row = new ArrayList<>();
        row.add(1);
        for(int col=1;col<n;col++){
            ans *= (n-col);
            ans /=col;
            row.add((int)ans);
        }
        return row;
    }
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> ans = new ArrayList<>();
        for(int i=1;i<=numRows;i++){
            ans.add(rows(i));
        }
        return ans;
    }
}